### PR TITLE
Preneed Schema

### PR DESCRIPTION
**Pull request by U-DON**

### CODE REVIEW

**Comment by U-DON** [comment]
@mphprogrammer Updated for multiple applicants; some things might still need minor tweaks to have the right `enum` values, but running for this for now to continue building out the actual form and then fixing it those minor issues later.

**Comment by jbalboni** [comment]
PR for issue 2 is here: https://github.com/department-of-veterans-affairs/vets-website/pull/6022

File: src/schemas/40-10007/schema.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          applicant: {
+            type: 'object',
+            properties: {
+              applicantEmail: { type: 'string', format: 'email' },
+              applicantPhoneNumber: schemaHelpers.getDefinition('phone'),
+              applicationRelationshipToClaimant: {
```
Comment (mphprogrammer) [inline src/schemas/40-10007/schema.js]: Should this (`applicationRelationshipToClaimant`) be `applicantRelationshipToClaimant`?

File: src/schemas/40-10007/schema.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                type: 'string',
+                'enum': [
+                  'self',
```
Comment (mphprogrammer) [inline src/schemas/40-10007/schema.js]: Assuming case sensitivity, I think `self` should be `Self`

File: src/schemas/40-10007/schema.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                  'Authorized Agent/Rep'
+                ]
+              },
+              completingReason: { type: 'string' },
+              mailingAddress: schemaHelpers.getDefinition('address'),
+              name: schemaHelpers.getDefinition('fullName')
```
Comment (mphprogrammer) [inline src/schemas/40-10007/schema.js]: EOAS has a `maidenName` field as well (currently max 15 characters, not required)

File: src/schemas/40-10007/schema.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+            }
+          },
+          claimant: {
+            type: 'object',
+            properties: {
+              address: schemaHelpers.getDefinition('address'),
+              dateOfBirth: schemaHelpers.getDefinition('date'),
+              desiredCemetery: { type: 'integer' },
+              email: { type: 'string', format: 'email' },
+              name: schemaHelpers.getDefinition('fullName'),
```
Comment (mphprogrammer) [inline src/schemas/40-10007/schema.js]: EOAS has a `maidenName` field as well (currently max 15 characters, not required)

File: src/schemas/40-10007/schema.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+              maidenName: { type: 'string' },
```
Comment (mphprogrammer) [inline src/schemas/40-10007/schema.js]: EOAS has a `maidenName` field as part of the name object for every `name`-type field.

File: src/schemas/40-10007/schema.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+                properties: {
+                  type: {
+                    type: 'integer',
+                    enum: [
+                      1, // Veteran
+                      2, // Spouse/Surviving Spouse
+                      3  // Unmarried Adult Child
+                    ]
+                  },
+                  other: { type: 'string' }
```
Comment (mphprogrammer) [inline src/schemas/40-10007/schema.js]: I think `other` wouldn't be necessary for relationshipToVet, not sure of its function.

Comment (mphprogrammer) [inline src/schemas/40-10007/schema.js]: could we leave it as an enum only, with ["1", "2", "3"]?

File: src/schemas/40-10007/schema.js [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+                }
+              },
+              ssn: schemaHelpers.getDefinition('ssn')
```
Comment (mphprogrammer) [inline src/schemas/40-10007/schema.js]: EOAS has it as \d{3}-\d{2}-\d{3}, but I can insert the dashes if you prefer.

**Comment by mphprogrammer** [comment]
@U-DON A bunch of stuff in the schema has changed. This should be a relatively stable verison: https://github.com/department-of-veterans-affairs/vets.gov-team/blob/master/Products/Pre-Needs-Burial/api/receive_preneeds_application_schema.json

**Pull request merged**